src/Couchbase.Extensions.Caching/CouchbaseCacheServiceCollectionExtensions.cs

```diff
@@ -29,5 +30,58 @@ public static IServiceCollection AddDistributedCouchbaseCache(this IServiceColle
 
             return services;
         }
+
+        /// <summary>
+        /// Adds a <see cref="CouchbaseCache"/> as a service using a <see cref="Action{CouchbaseCacheOptions}"/>
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the <see cref="CouchbaseCache"/> service to.</param>
+        /// <param name="bucketName">The bucket name that the cache will use.</param>
+        /// <param name="setupAction">The setup delegate that will be fired when the service is created.</param>
+        /// <returns>The <see cref="IServiceCollection"/> that was updated with the <see cref="Action{CouchbaseCacheOptions}"/></returns>
+        public static IServiceCollection AddDistributedCouchbaseCache(this IServiceCollection services, string bucketName, Action<CouchbaseCacheOptions> setupAction)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+            if (setupAction == null)
+            {
+                throw new ArgumentNullException(nameof(setupAction));
+            }
+
+            services.AddCouchbaseBucket<ICouchbaseCacheBucketProvider>(bucketName);
+            services.AddOptions();
+            services.Configure(setupAction);
+            services.Add(ServiceDescriptor.Singleton<IDistributedCache, CouchbaseCache>());
```

Comment:
This can also be written as `services.TryAddSingleton<IDistributedCache, CouchbaseCache>()`, which reads a bit better I think.

Also, TryAdd will not add the mapping if there is already a mapping present.  Add will add it anyway, resulting in a collection of implementations which is probably not desirable.  Using TryAdd will allow consumers to override the cache implementation with one inherited from CouchbaseCache, while still using the other features of this function.